components/application-registry/internal/metadata/applications/repository_test.go

```diff
@@ -236,6 +236,56 @@ func TestCreateServices(t *testing.T) {
 		assert.Equal(t, apperrors.CodeAlreadyExists, err.Code())
 	})
 
+	t.Run("should not allow to create service if service with the same display name already exists", func(t *testing.T) {
+		// given
+		application := createApplication("production")
+		appManagerMock := &mocks.AppManager{}
+		appManagerMock.On("Get", context.Background(), "production", metav1.GetOptions{}).
+			Return(application, nil)
+
+		repository := applications.NewServiceRepository(appManagerMock)
+		require.NotNil(t, repository)
+
+		newService := applications.Service{
+			ID:              "idx",
+			Name:            "",
+			DisplayName:     "Orders API",
+			LongDescription: "This is another Orders API",
+			Tags:            []string{"orders-second"},
+			Events:          true,
+		}
+		// when
+		err := repository.Create("production", newService)
+
+		// then
+		assert.Equal(t, apperrors.CodeWrongInput, err.Code())
+	})
+
+	t.Run("should not allow to create service if a service with the same normalized display name already exists", func(t *testing.T) {
+		// given
```

Comment:
Why do we need to test `display name` and `normalized display name` separately?

Reply:
I agree both cases, can be reduces into single one - for testing the normalized display name. I will change that.